### PR TITLE
libsignal-protocol-c 2.3.2 (new formula)

### PR DESCRIPTION
**Pull request by tehnick**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by tehnick** [comment]
Fixed.

**Comment by lembacon** [comment]
And a functional `test` is required for new formulae to get accepted. Please check out [Formula Cookbook](https://docs.brew.sh/Formula-Cookbook) 😉.

**Comment by tehnick** [comment]
See my [test](https://github.com/mxe/mxe/blob/master/src/libsignal-protocol-c-test.c) from `mxe` project for example.

**Comment by tehnick** [comment]
What is wrong with your jenkins tests? I do not see the reason of failure.

**Comment by tehnick** [comment]
Hmm, I have found:
```
Error: 1 problem in 1 formula
libsignal-protocol-c:
  * C: 1: col 1: A `test do` test block should be added
```
Is it really necessary even if it is useless?

**Comment by tehnick** [comment]
I have added test program. It is almost useless: we only check that few functions from library API will not be changed in new versions of library.

File: Formula/libsignal-protocol-c.rb [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      #include <signal_protocol.h>
+      #include <session_builder.h>
+      #include <session_cipher.h>
+      #include <stdio.h>
+      #include <string.h>
+
+      int main(void)
+      {
+        int result = 0;
+        printf("Beginning useless test...\\n");
```
Comment (DomT4) [inline Formula/libsignal-protocol-c.rb]: Let's avoid printing mildly sassy comments in `test do` blocks, thanks.

File: Formula/libsignal-protocol-c.rb [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+        session_cipher_free(cipher);
+        session_builder_free(builder);
+        signal_protocol_store_context_destroy(store_context);
+        printf("9\\n");
+
+        return 0;
+      }
+    EOS
+    system ENV.cc, "test.c", "-Wall", "-Werror", "-I#{include}/signal", "-L#{lib}", "-lsignal-protocol-c", "-o", "test"
```
Comment (DomT4) [inline Formula/libsignal-protocol-c.rb]: Leave out `"-Wall", "-Werror"` and ensure the rest is wrapped to 80 characters.

**Comment by DomT4** [comment]
> Is it really necessary even if it is useless?

We consider new formulae without any form of test inadmissible. It wouldn't be a hard audit failure if the requirement was flexible...

**Comment by tehnick** [comment]
Updated.

**Pull request closed without merging**